Source/Csla.AspNetCore/Blazor/ApplicationContextManagerBlazor.cs

```diff
@@ -63,21 +63,30 @@ public ApplicationContextManagerBlazor(IHttpContextAccessor hca, AuthenticationS
     /// </summary>
     protected HttpContext HttpContext { get; set; }
 
-    private void InitializeUser()
+    private async Task InitializeUser()
     {
       if (ActiveCircuitState.CircuitExists)
       {
         Task<AuthenticationState> task;
         try
         {
           task = AuthenticationStateProvider.GetAuthenticationStateAsync();
+          await task;
         }
         catch (InvalidOperationException ex)
         {
           task = Task.FromResult(new AuthenticationState(UnauthenticatedPrincipal));
           string message = ex.Message;
-          if (message.Contains(nameof(AuthenticationStateProvider.GetAuthenticationStateAsync))
-              && message.Contains(nameof(IHostEnvironmentAuthenticationStateProvider.SetAuthenticationState)))
+
+          //check for pre .net 8 error text
+          bool oldGetCalledBeforeSetError = message.Contains(nameof(AuthenticationStateProvider.GetAuthenticationStateAsync))
+            && message.Contains(nameof(IHostEnvironmentAuthenticationStateProvider.SetAuthenticationState));
+
+          //check for .net 8+ error text
+          bool newGetCalledOutsideRazorScopeError = message.Contains(nameof(AuthenticationStateProvider.GetAuthenticationStateAsync)
+            + " outside of the DI scope for a Razor component", StringComparison.OrdinalIgnoreCase);
```

Comment:
Having hard-coded english text seems problematic - I must think that this text will vary depending on the culture of the user and/or server?

Reply:
I wondered about that.  Any ideas?  The exception doesn't have anything else I can grab onto that I can see.

Reply:
Here is the exception in aspnet.core

https://github.com/dotnet/aspnetcore/blob/87e324a61dcd15db4086b8a8ca7bd74ca1e0a513/src/Components/Server/src/Circuits/ServerAuthenticationStateProvider.cs#L16

Reply:
Stepping back for a moment...I guess the risk of simply testing for `message.Contains(nameof(AuthenticationStateProvider.GetAuthenticationStateAsync)` is low because this method is only hit on CTOR so it's not like it would be firing repeatedly over and over.  What do you think about me removing everything but that on the check for the `InvalidOperationException`?

Reply:
wow, it is amazing that they accepted hard-coded text into the repo like that.

Why do we need to be aware of the specific text? Is there some way we can shield against future changes? This seems fragile to me.